pallets/rmrk-core/src/functions.rs

```diff
@@ -1,5 +1,194 @@
+	) -> sp_std::result::Result<(CollectionId, NftId), DispatchError> {
+		ensure!(max_recursions > 0, Error::<T>::TooManyRecursions);
+		NFTs::<T>::remove(collection_id, nft_id);
+		if let Some(kids) = Children::<T>::take(collection_id, nft_id) {
+			for child in kids {
+				<Self as Nft<T::AccountId, StringLimitOf<T>>>::burn_nft(
+					child.0,
+					child.1,
+					max_recursions - 1,
+				)?;
+			}
+		}
+		Ok((collection_id, nft_id))
+	}
+
+	fn send(
+		sender: T::AccountId,
+		collection_id: CollectionId,
+		nft_id: NftId,
+		new_owner: AccountIdOrCollectionNftTuple<T::AccountId, CollectionId, NftId>,
+		max_recursions: u32,
+	) -> sp_std::result::Result<(CollectionId, NftId), DispatchError> {
+		let mut sending_nft =
+			NFTs::<T>::get(collection_id, nft_id).ok_or(Error::<T>::NoAvailableNftId)?;
+		ensure!(sending_nft.rootowner == sender.clone(), Error::<T>::NoPermission);
+
+		match new_owner.clone() {
+			AccountIdOrCollectionNftTuple::AccountId(account_id) => {
+				// Remove previous parental relationship
+				if let AccountIdOrCollectionNftTuple::CollectionAndNftTuple(cid, nid) =
+					sending_nft.owner
+				{
+					if let Some(mut kids) = Children::<T>::take(cid, nid) {
+						kids.retain(|&kid| kid != (collection_id, nft_id));
+						Children::<T>::insert(cid, nid, kids);
+					}
+				}
+				sending_nft.rootowner = account_id.clone();
+			}
+			AccountIdOrCollectionNftTuple::CollectionAndNftTuple(cid, nid) => {
+				let recipient_nft = NFTs::<T>::get(cid, nid).ok_or(Error::<T>::NoAvailableNftId)?;
+				// Check if sending NFT is already a child of recipient NFT
+				ensure!(
+					!Pallet::<T>::is_x_descendent_of_y(cid, nid, collection_id, nft_id),
+					Error::<T>::CannotSendToDescendent
+				);
+
+				// Remove parent if exists: first we only care if the owner is a non-AccountId)
+				if let AccountIdOrCollectionNftTuple::CollectionAndNftTuple(cid, nid) =
+					sending_nft.owner
+				{
+					// second we only care if the parent has children (it should)
+					if let Some(mut kids) = Children::<T>::take(cid, nid) {
+						// third we only "retain" the other children
+						kids.retain(|&kid| kid != (collection_id, nft_id));
+						Children::<T>::insert(cid, nid, kids);
+					}
+				}
+				if sending_nft.rootowner != recipient_nft.rootowner {
+					// sending_nft.rootowner = recipient_nft.rootowner
+					sending_nft.rootowner = recipient_nft.rootowner.clone();
+
+					Pallet::<T>::recursive_update_rootowner(
+						collection_id,
+						nft_id,
+						recipient_nft.rootowner,
+						max_recursions,
+					)?;
+				}
+				match Children::<T>::take(cid, nid) {
+					None => Children::<T>::insert(cid, nid, vec![(collection_id, nft_id)]),
+					Some(mut kids) => {
+						kids.push((collection_id, nft_id));
+						Children::<T>::insert(cid, nid, kids);
+					}
+				}
+			}
+		};
+		sending_nft.owner = new_owner.clone();
+
+		NFTs::<T>::remove(collection_id, nft_id);
+		NFTs::<T>::insert(collection_id, nft_id, sending_nft);
```

Comment:
```suggestion
		NFTs::<T>::insert(collection_id, nft_id, sending_nft);
```

An insert override the remove operation. So we don't need an additional remove here.